core/src/main/java/de/danoeh/antennapod/core/service/playback/PlaybackService.java

```diff
@@ -677,6 +677,11 @@ public void statusChanged(PlaybackServiceMediaPlayer.PSMPInfo newInfo) {
                     writePlayerStatusPlaybackPreferences();
                     break;
 
+                case STOPPED:
+                    writePlaybackPreferencesNoMediaPlaying();
+                    stopSelf();
```

Comment:
It looks like this commit, adding 'stopSelf()` for `case STOPPED`, intrdouced a bug. After finishing play the last item on the queue, AntennaPod crashed.

Relevant Log: 
```
06-09 10:22:57.379 F/art     (22928): art/runtime/runtime.cc:465] JNI DETECTED ERROR IN APPLICATION: JNI NewObjectV called with pending exception java.lang.IllegalStateException: 
06-09 10:22:57.379 F/art     (22928): art/runtime/runtime.cc:465]   at android.media.PlaybackParams android.media.MediaPlayer.getPlaybackParams() (MediaPlayer.java:-2)
06-09 10:22:57.379 F/art     (22928): art/runtime/runtime.cc:465]   at void org.antennapod.audio.AndroidAudioPlayer.setPitchStepsAdjustment(float) (AndroidAudioPlayer.java:376)
06-09 10:22:57.379 F/art     (22928): art/runtime/runtime.cc:465]   at void org.antennapod.audio.MediaPlayer.switchMediaPlayerImpl(org.antennapod.audio.AbstractAudioPlayer, org.antennapod.audio.AbstractAudioPlayer) (MediaPlayer.java:542)
06-09 10:22:57.379 F/art     (22928): art/runtime/runtime.cc:465]   at void org.antennapod.audio.MediaPlayer.setupMpi(android.content.Context) (MediaPlayer.java:478)
06-09 10:22:57.379 F/art     (22928): art/runtime/runtime.cc:465]   at void org.antennapod.audio.MediaPlayer.fallback() (MediaPlayer.java:361)
06-09 10:22:57.379 F/art     (22928): art/runtime/runtime.cc:465]   at boolean de.danoeh.antennapod.core.service.playback.LocalPSMP.lambda$new$23$LocalPSMP(org.antennapod.audio.MediaPlayer, int, int) (LocalPSMP.java:929)
06-09 10:22:57.379 F/art     (22928): art/runtime/runtime.cc:465]   at boolean de.danoeh.antennapod.core.service.playback.LocalPSMP$$Lambda$7.onError(org.antennapod.audio.MediaPlayer, int, int) ((null):-1)
06-09 10:22:57.379 F/art     (22928): art/runtime/runtime.cc:465]   at void org.antennapod.audio.SonicAudioPlayer.error(int) (SonicAudioPlayer.java:474)
06-09 10:22:57.379 F/art     (22928): art/runtime/runtime.cc:465]   at void org.antennapod.audio.SonicAudioPlayer.error() (SonicAudioPlayer.java:465)
06-09 10:22:57.379 F/art     (22928): art/runtime/runtime.cc:465]   at void org.antennapod.audio.SonicAudioPlayer.stop() (SonicAudioPlayer.java:240)
06-09 10:22:57.379 F/art     (22928): art/runtime/runtime.cc:465]   at void org.antennapod.audio.MediaPlayer.stop() (MediaPlayer.java:1269)
06-09 10:22:57.380 F/art     (22928): art/runtime/runtime.cc:465]   at void de.danoeh.antennapod.core.service.playback.LocalPSMP.shutdown() (LocalPSMP.java:613)
06-09 10:22:57.380 F/art     (22928): art/runtime/runtime.cc:465]   at void de.danoeh.antennapod.core.service.playback.PlaybackService.onDestroy() (PlaybackService.java:362)
^^^ In  mediaPlayer.shutdown() of onDestroy()
```

Reply:
I think this bug was introduced by me when I added `mediaPlayer.stop();` to `LocalPSMP.shutdown`. This was added to prevent the playback to be continued in background without a way to control it. I added a commit that fixes this.